portafly/src/components/LazyRoute.tsx

```diff
@@ -2,19 +2,10 @@ import * as React from 'react'
 import { RouteProps, Route } from 'react-router-dom'
 import { Loading } from 'components'
 
-export interface IDynamicImportProps extends RouteProps {
-  getComponent: () => Promise<{ default: React.ComponentType }>
-}
-
-export function LazyRoute({ path, exact, getComponent }: IDynamicImportProps) {
-  const LazyComponent = React.useMemo(() => React.lazy(getComponent), [
```

Comment:
So we are losing memoization?

Reply:
It didn't make sense in the first place AFAIK.

Reply:
In this case it seems it's good it seems. While visiting each page, you are lazy loading them and memoizing, second time you visit that page, and the props haven't changed, it's super fast.

Reply:
The theory is valid but `useMemo` doesn't work like that. It is meant to save in _expensive_ computations (which we don't have) when **re-rendering** a component, but we aren't re-rendering either. The React.lazy trick will have in the DOM only one Route at a time, therefore when visiting a second time a page, it will render from scratch. Which again, it's super fast anyway.

You can check it yourself by adding a log:
```jsx
const LazyComponent = React.useMemo(() => {
    console.log('useMemo')
    return React.lazy(getComponent)
  }, [getComponent])
```